File: tests/csharp/tests-csharp-xunit-api_request-playwright/Tests/Ui/HomeErrorStateTests.cs
using Allure.Net.Commons;
using Allure.Net.Commons.Attributes;
using Api;
using Tests;

namespace Tests.Ui;

[AllureLabel("layer", "ui")]
[AllureEpic("Home")]
[AllureFeature("Error states")]
[AllureSeverity(SeverityLevel.normal)]
[AllureSuite("Home error states (mock)")]
public sealed class HomeErrorStateTests : TestBase
{
    private readonly bool _mockStandAvailable = true;

    public HomeErrorStateTests() : this(EnsureMockStand())
    {
    }

    private HomeErrorStateTests(bool _)
    {
    }

    /// <summary>Runs before <see cref="TestBase"/> so a missing mock stand does not start Chromium.</summary>
    private static bool EnsureMockStand()
    {
        if (!MockScenarios.Available())
        {
            throw Xunit.Sdk.SkipException.ForSkip(
                "WireMock admin API is not exposed on this stand — error injection needs the mock profile");
        }

        return true;
    }

    public override void Dispose()
    {
        if (_mockStandAvailable)
        {
            MockScenarios.ResetAll();
        }

        base.Dispose();
    }

    [Trait("TestCategory", "ui")]
    [Trait("TestCategory", "mock")]
    [Trait("TestCategory", "negative")]
    [Fact(DisplayName = "Items API failure shows a readable error, not a blank page")]
    public void ItemsApiFailureShowsReadableError()
    {
        MockScenarios.SetState("items", "error");
        HomePage.OpenPage()
            .ShouldShowItemsError("✗ items: HTTP 500");
    }

    [Trait("TestCategory", "ui")]
    [Trait("TestCategory", "mock")]
    [Trait("TestCategory", "negative")]
    [Fact(DisplayName = "Health API failure shows a readable error in the health panel")]
    public void HealthApiFailureShowsReadableError()
    {
        MockScenarios.SetState("health", "error");
        HomePage.OpenPage()
            .ShouldShowHealthError("✗ health: HTTP 500");
    }
}

File: tests/kotlin/tests-kotlin-junit5-api_request-playwright/src/test/kotlin/tests/ui/HomeErrorStateTests.kt
package tests.ui

import annotations.Layer
import api.MockScenarios
import com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat
import io.qameta.allure.Epic
import io.qameta.allure.Feature
import io.qameta.allure.Severity
import io.qameta.allure.SeverityLevel
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assumptions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.parallel.Execution
import org.junit.jupiter.api.parallel.ExecutionMode
import org.junit.jupiter.api.parallel.Isolated
import tests.TestBase

@Layer("ui")
@Epic("Home")
@Feature("Error states")
@Severity(SeverityLevel.NORMAL)
@DisplayName("Home error states (mock)")
@Isolated("WireMock scenarios are process-global")
@Execution(ExecutionMode.SAME_THREAD)
class HomeErrorStateTests : TestBase() {

    private var mockStandAvailable = false

    @BeforeEach
    fun requireMockStand() {
        mockStandAvailable = MockScenarios.available()
        Assumptions.assumeTrue(
            mockStandAvailable,
            "WireMock admin API is not exposed on this stand — error injection needs the mock profile",
        )
    }

    @AfterEach
    fun resetScenarios() {
        if (mockStandAvailable) {
            MockScenarios.resetAll()
        }
    }

    @Test
    @Tag("ui")
    @Tag("mock")
    @Tag("negative")
    @DisplayName("Items API failure shows a readable error, not a blank page")
    fun itemsApiFailureShowsReadableError() {
        MockScenarios.setState("items", "error")
        app.home.open()
        assertThat(app.home.itemsList).containsText("✗ items: HTTP 500")
    }

    @Test
    @Tag("ui")
    @Tag("mock")
    @Tag("negative")
    @DisplayName("Health API failure shows a readable error in the health panel")
    fun healthApiFailureShowsReadableError() {
        MockScenarios.setState("health", "error")
        app.home.open()
        assertThat(app.home.healthStatus).containsText("✗ health: HTTP 500")
    }
}
